enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.ts

```diff
@@ -25,18 +23,38 @@ interface InitDataLoaderParameters {
 export const useInitData = ({ config }: InitDataLoaderParameters) => {
   const { allowConsoleLog = true } = config;
 
+  // react calls some lifecycle hooks twice in dev mode, the auth init fires some http requests and when it's called twice,
+  // it fires them twice as well, making debugging harder as they show up twice in the network tab and in the logs
+  const hasBeenInitialized = useRef(false);
+
+  const dispatch = useSdkDispatch();
+
+  const loginStatus = useSdkSelector(getLoginStatus);
+  const fetchRefreshTokenFnFromStore = useSdkSelector(getFetchRefreshTokenFn);
+
   // This is outside of a useEffect otherwise calls done on the first render could use the wrong value
   // This is the case for example for the locale json files
   if (api.basename !== config.metabaseInstanceUrl) {
     api.basename = config.metabaseInstanceUrl;
   }
+  if (config.fetchRequestToken !== fetchRefreshTokenFnFromStore) {
```

Comment:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Reply:
This looks weird IMO. We shouldn't allow setting `null`, or set `null` as a function value, can we have the default function set and only allow to override it if it's set to something else not empty?

Similar to https://github.com/metabase/metabase/pull/49214#discussion_r1824399615

```ts
if (config.fetchRequestToken != null && config.fetchRequestToken !== fetchRefreshTokenFnFromStore) {
  dispatch(setFetchRefreshTokenFn(config.fetchRequestToken));
}
```

Reply:
Answered in the other thread, but tldr: the default value of `fetchRefreshTokenFnFromStore` is null so we are not changing the logic